src/Client.ts

```diff
@@ -703,3 +628,33 @@ async function getUserContactsFromNetwork(
     })
   )
 }
+
+export function defaultKeystoreProviders(): KeystoreProvider[] {
+  return [
+    // First check to see if a `privateKeyOverride` is provided and use that
+    new StaticKeystoreProvider(),
+    // Next check to see if a EncryptedPrivateKeyBundle exists on the network for the wallet
+    new NetworkKeystoreProvider(),
+    // If the first two failed with `KeystoreProviderUnavailableError`, then generate a new key and write it to the network
+    new KeyGeneratorKeystoreProvider(),
```

Comment:
I really this pattern but does it mean that transient errors with NetworkKeystoreProvider could cause a new identity key + pre key bundle to be created?

Reply:
Oh I guess the code below contains a partial answer: the bootstrap process will only continue if a KeystoreProviderUnavailableError is thrown. That shouldn't happen if it's a transient 404 on a topic right?

Reply:
Great question. My goal was to replicate the behaviour today.

If the network has an error and does not return a valid response, that error will bubble and the bootstrapping process will abort.

But if the network returns a valid empty response we will create a new bundle and publish it.

Reply:
Awesome, thanks!